frontend/javascripts/oxalis/shaders/segmentation.glsl.js

```diff
@@ -7,29 +7,167 @@ import { hsvToRgb } from "./utils.glsl";
           <%= segmentationName %>_mapping_color_texture,
           <%= mappingColorTextureWidth %>,
           0.0
         ).r != -1.0;
         if (isMappingEnabled && hasCustomMappingColors) {
-          value = getRgbaAtIndex(
+          colorValue = getRgbaAtIndex(
             <%= segmentationName %>_mapping_color_texture,
             <%= mappingColorTextureWidth %>,
             lastEightBits
           ).r;
         }
       <% } %>
 
-      vec4 HSV = vec4( value, 1.0, 1.0, 1.0 );
+      // The following code scales the world coordinates so that the coordinate frequency is in a "pleasant" range.
+      // Also, when zooming out, coordinates change faster which make the pattern more turbulent. Dividing by the
+      // zoomValue compensates this. Note that the zoom *step* should not be used here, because that value relates to the
+      // three-dimensional dataset. Since the patterns are only 2D, the zoomValue is a better proxy.
+      //
+      // By default, scale everything with fineTunedScale as this seemed a good value during testing.
+      float fineTunedScale = 0.15;
+      // Additionally, apply another scale factor (between 0.5 and 1.5) depending on the segment id.
+      float frequencySequenceLength = 3.;
+      float frequencyModulator = mix(0.5, 1.5, getElementOfPermutation(significantSegmentIndex, frequencySequenceLength, 2.) / frequencySequenceLength);
+      float coordScaling = fineTunedScale * frequencyModulator;
+      // Round the zoomValue so that the pattern frequency only changes at distinct steps. Otherwise, zooming out
+      // wouldn't change the pattern at all, which would feel weird.
+      float zoomAdaption = ceil(zoomValue);
+      vec3 worldCoordUVW = coordScaling * getWorldCoordUVW()  / zoomAdaption;
+
+      float baseVoxelSize = min(min(datasetScale.x, datasetScale.y), datasetScale.z);
+      vec3 datasetScaleUVW = transDim(datasetScale) / baseVoxelSize;
+      worldCoordUVW.x = worldCoordUVW.x * datasetScaleUVW.x;
+      worldCoordUVW.y = worldCoordUVW.y * datasetScaleUVW.y;
+
+      float angleCount = 17.;
+      float angle = 1.0 / angleCount * getElementOfPermutation(significantSegmentIndex, angleCount, 3.0);
+
+      float stripeValueA = mix(
+        worldCoordUVW.x,
+        worldCoordUVW.y,
+        angle
+      );
+      float stripeValueB = mix(
+        worldCoordUVW.x,
+        -worldCoordUVW.y,
+        1.0 - angle
+      );
+
+      // useGrid is binary, but we generate a pseudo-random sequence of 13 elements which we map
+      // to ones and zeros. This has the benefit that the periodicity has a prime length.
+      float useGridSequenceLength = 13.;
+      float useGrid = step(mod(getElementOfPermutation(significantSegmentIndex, useGridSequenceLength, 2.0), 2.0), 0.5);
+      float aaStripeValueA = aa_step(stripeValueA);
+      float aaStripeValueB = aa_step(stripeValueB);
+      float aaStripeValue = 1.0 - max(aaStripeValueA, useGrid * aaStripeValueB);
```

Comment:
The two stripes A and B are orthogonal to each other. I `useGrid` is true, these orthogonal stripes are combined (by taking the max of both), so that a grid is produced. If `useGrid` is false, only the first stripe A is used (so, there is a simple stripe pattern).
I'm not sure how to make the naming more clear, but I'll add an explanation to the two stripe value definitions so that this is easier to understand.

Reply:
> The two stripes A and B are orthogonal to each other. I useGrid is true, these orthogonal stripes are combined (by taking the max of both), so that a grid is produced. If useGrid is false, only the first stripe A is used (so, there is a simple stripe pattern).

Wow thats really smart :man_student: